src/test/java/uk/gov/companieshouse/registeredemailaddressapi/unit/service/EligibilityServiceTest.java

```diff
@@ -0,0 +1,169 @@
+import uk.gov.companieshouse.registeredemailaddressapi.service.PrivateDataRetrievalService;
+
+@ExtendWith(MockitoExtension.class)
+class EligibilityServiceTest {
+
+
+    private static final String COMPANY_NUMBER = "12345";
+
+    private List<EligibilityRule<CompanyProfileApi>> eligibilityRules;
+
+    @Mock
+    private CompanyProfileService companyProfileService;
+
+    @Mock
+    private PrivateDataRetrievalService privateDataRetrievalService;
+
+    private EligibilityService eligibilityService;
+
+    @BeforeEach
+    void init() {
+        CompanyTypeValidation companyTypeValidation = new CompanyTypeValidation(Set.of("ltd"));
+        CompanyStatusValidation companyStatusValidation = new CompanyStatusValidation(Set.of("active"));
+        CompanyEmailValidation companyEmailValidation = new CompanyEmailValidation(privateDataRetrievalService);
+
+        eligibilityRules = List.of(companyTypeValidation, companyStatusValidation, companyEmailValidation);
+
+        eligibilityService = new EligibilityService(eligibilityRules, companyProfileService);
+    }
+
+    @Test
+    void testWithNoErrors() throws ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyNumber(COMPANY_NUMBER);
+        companyProfileApi.setCompanyStatus("active");
+        companyProfileApi.setType("ltd");
+
+        RegisteredEmailAddressJson registeredEmailAddressJson = new RegisteredEmailAddressJson();
+        registeredEmailAddressJson.setRegisteredEmailAddress("info@acme.com");
+
+        BDDMockito.given(privateDataRetrievalService.getRegisteredEmailAddress(COMPANY_NUMBER)).willReturn(registeredEmailAddressJson);
+
+        // WHEN
+        var responseBody = eligibilityService.checkCompanyEligibility(companyProfileApi);
+
+        // THEN
+        assertNotNull(responseBody);
+        assertEquals(EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE, responseBody.getEligibilityStatusCode());
+    }
+
+    @Test
+    void testInvalidCompanyStatus() throws EligibilityException, ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyNumber(COMPANY_NUMBER);
+        companyProfileApi.setCompanyStatus("inactive");
+        companyProfileApi.setType("ltd");
+
+        // WHEN
+        var responseBody = eligibilityService.checkCompanyEligibility(companyProfileApi);
+
+        // THEN
+        assertNotNull(responseBody);
+        assertEquals(EligibilityStatusCode.INVALID_COMPANY_STATUS, responseBody.getEligibilityStatusCode());
+    }
+
+    @Test
+    void testInvalidCompanyType() throws EligibilityException, ServiceException {
+        // GIVEN
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyNumber(COMPANY_NUMBER);
+        companyProfileApi.setCompanyStatus("active");
+        companyProfileApi.setType("plc");
```

Comment:
Is this an invalid company type?

Reply:
In the context of the test it is invalid (as per construction of the CompanyTypeValidation in @BeforeEach method) but for extra clarity have changed to using a invalid company type not specified in application.properties/allowed.company.types